pycrostates/preprocessing/extract_gfp_peaks.py

```diff
@@ -27,10 +27,24 @@
 if TYPE_CHECKING:
     from typing import Optional, Union
 
-    from .._typing import Picks, ScalarFloatArray
+    from .._typing import Picks
     from ..io import ChData
 
 
+def _std():
+    def compute_std(data):
+        return np.std(data, axis=0)
+
+    return compute_std
+
+
+def _rms():
+    def compute_rms(data):
+        return np.sqrt(np.mean(data**2, axis=0))
+
+    return compute_rms
```

Comment:
No need to nest here and you could use a dict to store the GFP functions per channel type:

```python
_GFP_FUNC: dict[str, Callable] = {
    "eeg": np.std,
    "grad": lambda x: np.sqrt(np.mean(x**2, axis=0)),
    "mag": lambda x: np.sqrt(np.mean(x**2, axis=0)),
}


_check_value(ch_type, _GFP_FUNC, "ch_type")
gfp = _GFP_FUNC[ch_type](data)
```

It removes the additional private functions and the nesting of function within, and removes the if/else logic on channel types.